packages/hub/src/hub.ts

```diff
@@ -148,10 +148,10 @@ export class Hub implements HubInterface {
   /**
    * @inheritDoc
    */
-  public withScope(callback: (scope: Scope) => void): void {
+  public withScope<T>(fn: (scope: Scope) => T): T {
     const scope = this.pushScope();
     try {
-      callback(scope);
+      return fn(scope);
     } finally {
       this.popScope();
```

Comment:
@kamilogorek noted well yesterday that [`withScope`'s existing behavior swallows exceptions](https://github.com/getsentry/sentry-javascript/blob/7246a22a6bce6cb41e1f7e899d12fddefde68c49/packages/hub/src/hub.ts#L151-L158) thrown within the callback function (and the same is true for [`Hub.run`](https://github.com/getsentry/sentry-javascript/blob/7246a22a6bce6cb41e1f7e899d12fddefde68c49/packages/hub/src/hub.ts#L351-L358)).

If we keep that behavior, a function call wrapped with `withScope` would, independent of the declared return type, return `undefined` instead of throwing.

That behavior is not desirable, thus we should not build `Sentry.trace` on top of `withScope` unless we make it re-throw. That's a bigger behavior change, and it may reveal exceptions in existing code that are being swallowed.

Needs a decision. @mitsuhiko thoughts?

Reply:
Disregard, `try...finally` doesn't swallow exceptions, that was a mistake 😓